### PR TITLE
Updating checkbox to handle tap

### PR DESCRIPTION
**Pull request by kevinDespoulains**
Hi,

Linked to issue [#158](https://github.com/singerdmx/flutter-quill/issues/158), I was looking for a way to make checkbox clickable.

What to you think about this?

### CODE REVIEW

File: lib/widgets/text_block.dart [hunk 1]
```diff
@@ -161,12 +163,23 @@ class EditableTextBlock extends StatelessWidget {
 
     if (attrs[Attribute.list.key] == Attribute.checked) {
       return _Checkbox(
-          style: defaultStyles!.leading!.style, width: 32, isChecked: true);
+        key: UniqueKey(),
```
Comment (matuella) [inline lib/widgets/text_block.dart:166]: Any particular reason for adding an `UniqueKey` every time this widget builds?

File: lib/widgets/text_block.dart [hunk 2]
```diff
@@ -685,46 +698,41 @@ class _BulletPoint extends StatelessWidget {
   }
 }
 
-class _Checkbox extends StatefulWidget {
-  const _Checkbox({Key? key, this.style, this.width, this.isChecked})
-      : super(key: key);
-
+/// A Checkbox widget.
+class _Checkbox extends StatelessWidget {
+  /// Creates a [_Checkbox].
```
Comment (matuella) [inline lib/widgets/text_block.dart:703]: I don't think these docs add any relevant info to the class, personally I would remove them or get a little more in-depth of what they actually do.

**Review by matuella — COMMENTED** [review]
That looks like a good solution for me, I've tested it and apparently works fine. Ideally, we wouldn't need a gesture detector to solve this problem, as the checkbox itself knows how to handle this, but I'm not sure the root of this problem (maybe conflicting gestures?), so I would say this is overall good. I can't approve this because you've removed some code that I am not sure the impact of it (the `_flipListCheckbox` one).

@singerdmx, do you know if the removed code makes sense?

File: lib/widgets/text_block.dart (hunk 1 above)
Comment (kevinDespoulains) [reply lib/widgets/text_block.dart:166]: Yes, without using a key you can observe an unwanted animation on checkboxes when first chekbox is tapped.

Tested on version 1.2.1 :
Animation

File: lib/widgets/text_block.dart (hunk 2 above)
Comment (kevinDespoulains) [reply lib/widgets/text_block.dart]: I removed these lines in the latest commit.

**Comment by kevinDespoulains** [comment]
- The GestureDetector is used to handle long press, as suggested in issue https://github.com/singerdmx/flutter-quill/issues/158. I could not find a way to handle this gesture directly from checkbox.
- For the _flipListCheckbox part, I removed it to avoid checking/unchecking a checkbox on a click next to it (like it was before). Let me know if you want me to add it back.

File: lib/widgets/text_block.dart (hunk 1 above)
Comment (GyuriMajercsik) [reply lib/widgets/text_block.dart:166]: A `Key(index)` would make more sense. Basically, each item should have its own key.

Comment (matuella) [reply lib/widgets/text_block.dart:166]: I agree that a `Key` is necessary here, as they are all of the same class type, but just like @GyuriMajercsik suggested, a key that doesn't change every time this is built would be ideal. While IMO [index is not ideal](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318), it's probably decent for this particular case here.

**Comment by matuella** [comment]
> * For the _flipListCheckbox part, I removed it to avoid checking/unchecking a checkbox on a click next to it (like it was before). Let me know if you want me to add it back.

Apparently that's fine for me, just wanted to make sure that @singerdmx also thinks this as well.

**Pull request merged**